pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyClientCnx.java

```diff
@@ -66,7 +65,7 @@ protected ByteBuf newConnectCommand() throws Exception {
         authenticationDataProvider = authentication.getAuthData(remoteHostName);
         AuthData authData = authenticationDataProvider.authenticate(AuthData.INIT_AUTH_DATA);
         return Commands.newConnect(authentication.getAuthMethodName(), authData, protocolVersion,
-                PulsarVersion.getVersion(), proxyToTargetBrokerAddress, clientAuthRole, clientAuthData,
+                proxyConnection.clientVersion, proxyToTargetBrokerAddress, clientAuthRole, clientAuthData,
```

Comment:
This change is incorrect. When we use the `ProxyClientCnx`, the proxy uses its own version of the protocol with the broker, not the original client's version.

Reply:
The `clientVersion` here is different from the `protocol version`. There is another field called `protoclVersion` here:
https://github.com/apache/pulsar/blob/954f406fab03ce8858883335ce6d5a676bb45c92/pulsar-common/src/main/proto/PulsarApi.proto#L273
The proxy will choose the lowest protocol version for itself and the client:
https://github.com/apache/pulsar/blob/954f406fab03ce8858883335ce6d5a676bb45c92/pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java#L685-L687

Reply:
Yes, I am familiar with the difference for the protocol and the client version. However, it seems debatable to me which client version is connected here because the proxy is re-writing all requests instead of just forwarding them. For the purpose of troubleshooting, it seems valuable to know that the proxy is in the middle for these lookup connections.